File: models/identitas_proyek.go
package models

import "time"

type IdentitasProyek struct {
	ID                  uint      `gorm:"primaryKey" json:"id"`
	Nama                *string   `json:"nama"`
	TahunAnggaran       *string   `json:"tahun_anggaran"`
	Kategori            *string   `json:"kategori"`
	Provinsi            *string   `json:"provinsi"`
	Kabupaten           *string   `json:"kabupaten"`
	Kecamatan           *string   `json:"kecamatan"`
	Kelurahan           *string   `json:"kelurahan"`
	Latitude            *string   `json:"latitude"`
	Longitude           *string   `json:"longitude"`
	NilaiKontrak        *string   `json:"nilai_kontrak"`
	KontraktorPelaksana *string   `json:"kontraktor_pelaksana"`
	KonsultasPengawas   *string   `json:"konsultas_pengawas"`
	SumberDana          *string   `json:"sumber_dana"`
	KontrakFile         *string   `json:"kontrak_file"`
	SuratPerintahFile   *string   `json:"surat_perintah_file"`
	SuratPenunjukanFile *string   `json:"surat_penunjukan_file"`
	BeritaAcaraFile     *string   `json:"berita_acara_file"`
	CreatedById         *uint     `json:"created_by_id"`
	CreatedAt           time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt           time.Time `gorm:"autoUpdateTime" json:"updated_at"`

	CreatedBy *User                      `gorm:"foreignKey:CreatedById" json:"created_by"`
	Photos    []*IdentitasProyekPhoto    `gorm:"foreignKey:identitas_proyek_id" json:"photos"`
	Documents []*IdentitasProyekDocument `gorm:"foreignKey:identitas_proyek_id" json:"documents"`
}

File: app/src/main/java/com/example/sipamas_android/data/model/IdentitasProyek.kt
package com.example.sipamas_android.data.model

data class IdentitasProyek(
    val id: Int? = null,
    val nama: String? = null,
    val tahun_anggaran: String? = null,
    val kategori: String? = null,
    val provinsi: String? = null,
    val kabupaten: String? = null,
    val kecamatan: String? = null,
    val kecamatan_kode: String? = null,
    val kelurahan: String? = null,
    val latitude: String? = null,
    val longitude: String? = null,
    val nilai_kontrak: String? = null,
    val kontraktor_pelaksana: String? = null,
    val konsultas_pengawas: String? = null,
    val sumber_dana: String? = null,
    val kontrak_file: String? = null,
    val surat_perintah_file: String? = null,
    val surat_penunjukan_file: String? = null,
    val berita_acara_file: String? = null,
    val alasan_count: Int? = null,
    val alasan_text: String? = null,
    val created_by_id: Int? = null,
    val created_at: String? = null,
    val updated_at: String? = null,
    val created_by: User? = null
)
